server/src/main/java/com/hedera/block/server/BlockStreamService.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Hedera Block Node
+ *
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.block.server;
+
+import com.google.protobuf.Descriptors;
+import com.hedera.block.protos.BlockStreamServiceGrpcProto;
+import com.hedera.block.server.producer.ProducerBlockStreamObserver;
+import com.hedera.block.server.mediator.StreamMediator;
+import com.hedera.block.server.consumer.LiveStreamObserverImpl;
+import com.hedera.block.server.consumer.LiveStreamObserver;
+import io.grpc.stub.StreamObserver;
+import io.helidon.webserver.grpc.GrpcService;
+
+import java.util.logging.Logger;
+
+import static com.hedera.block.server.Constants.*;
+
+/**
+ * This class implements the GrpcService interface and provides the functionality for the BlockStreamService.
+ * It sets up the bidirectional streaming methods for the service and handles the routing for these methods.
+ * It also initializes the StreamMediator, BlockStorage, and BlockCache upon creation.
+ *
+ * <p>The class provides two main methods, streamSink and streamSource, which handle the client and server streaming
+ * respectively. These methods return custom StreamObservers which are used to observe and respond to the streams.
+ *
+ */
+public class BlockStreamService implements GrpcService {
+
+    private final Logger logger = Logger.getLogger(getClass().getName());
+
+    private final long timeoutThresholdMillis;
+    private final StreamMediator<BlockStreamServiceGrpcProto.Block, BlockStreamServiceGrpcProto.BlockResponse> streamMediator;
+
+    /**
+     * Constructor for the BlockStreamService class.
+     *
+     * @param timeoutThresholdMillis the timeout threshold in milliseconds
+     * @param streamMediator the stream mediator
+     */
+    public BlockStreamService(long timeoutThresholdMillis,
```

Comment:
We use `final` liberally, including in method args. Also we use @NonNull / @Nullable annotations on all args and all methods with non-null Object return values. There is a version we use in services from FindBugs (`edu.umd.cs.findbugs.annotations.NonNull` for example).

Reply:
I've been looking at the `hedera-services` code to enable `edu.umd.cs.findbugs.annotations.NonNull`.  It's difficult to follow how this is achieved through Gradle.  The java classes have this dependency but the module-info.java is using `requires static com.github.spotbugs.annotations;`?  It looks like there's repackaging of annotations, etc?

Reply:
The umd.edu annotation is published in the spotbugs annotations module.  It's an historical anomaly related to a JEP that didn't get completed.

Reply:
I added a new ticket to add support for these annotations:  https://github.com/hashgraph/hedera-block-node/issues/30